docs/source/reference/nn.rst

```diff
@@ -3,6 +3,190 @@
 tensordict.nn package
 =====================
 
+The tensordict.nn package makes it possible to flexibly use TensorDict within
```

Comment:
@tcbegley can you have a quick look at this?
Some parts have been copy-pasted / simplified from out recent post